plugins/Synchro/SynchroSynth.cpp

```diff
@@ -0,0 +1,565 @@
+{
+	// Save plugin version (does this ever get used?)
+	thisElement.setAttribute("version", SYNCHRO_VERSION);
+
+	m_modulation.saveSettings(doc, thisElement, "modulation");
+	m_harmonics.saveSettings(doc, thisElement, "harmonics");
+	m_modulationStrength.saveSettings(doc, thisElement, "modulationStrength");
+
+	m_carrierDetune.saveSettings(doc, thisElement, "carrierDetune");
+	m_carrierDrive.saveSettings(doc, thisElement, "carrierDrive");
+	m_carrierSync.saveSettings(doc, thisElement, "carrierSync");
+	m_carrierChop.saveSettings(doc, thisElement, "carrierChop");
+	m_carrierAttack.saveSettings(doc, thisElement, "carrierAttack");
+	m_carrierDecay.saveSettings(doc, thisElement, "carrierDecay");
+	m_carrierSustain.saveSettings(doc, thisElement, "carrierSustain");
+	m_carrierRelease.saveSettings(doc, thisElement, "carrierRelease");
+
+	m_modulatorDetune.saveSettings(doc, thisElement, "modulatorDetune");
+	m_modulatorDrive.saveSettings(doc, thisElement, "modulatorDrive");
+	m_modulatorSync.saveSettings(doc, thisElement, "modulatorSync");
+	m_modulatorChop.saveSettings(doc, thisElement, "modulatorChop");
+	m_modulatorAttack.saveSettings(doc, thisElement, "modulatorAttack");
+	m_modulatorDecay.saveSettings(doc, thisElement, "modulatorDecay");
+	m_modulatorSustain.saveSettings(doc, thisElement, "modulatorSustain");
+	m_modulatorRelease.saveSettings(doc, thisElement, "modulatorRelease");
+}
+
+void SynchroSynth::loadSettings(const QDomElement & thisElement)
+{
+	m_modulation.loadSettings(thisElement, "modulation");
+	m_harmonics.loadSettings(thisElement, "harmonics");
+	m_modulationStrength.loadSettings(thisElement, "modulationStrength");
+
+	m_carrierDetune.loadSettings(thisElement, "carrierDetune");
+	m_carrierDrive.loadSettings(thisElement, "carrierDrive");
+	m_carrierSync.loadSettings(thisElement, "carrierSync");
+	m_carrierChop.loadSettings(thisElement, "carrierChop");
+	m_carrierAttack.loadSettings(thisElement, "carrierAttack");
+	m_carrierDecay.loadSettings(thisElement, "carrierDecay");
+	m_carrierSustain.loadSettings(thisElement, "carrierSustain");
+	m_carrierRelease.loadSettings(thisElement, "carrierRelease");
+
+	m_modulatorDetune.loadSettings(thisElement, "modulatorDetune");
+	m_modulatorDrive.loadSettings(thisElement, "modulatorDrive");
+	m_modulatorSync.loadSettings(thisElement, "modulatorSync");
+	m_modulatorChop.loadSettings(thisElement, "modulatorChop");
+	m_modulatorAttack.loadSettings(thisElement, "modulatorAttack");
+	m_modulatorDecay.loadSettings(thisElement, "modulatorDecay");
+	m_modulatorSustain.loadSettings(thisElement, "modulatorSustain");
+	m_modulatorRelease.loadSettings(thisElement, "modulatorRelease");
+}
+
+QString SynchroSynth::nodeName() const { return synchro_plugin_descriptor.name; };
+
+//Update carrier waveform view
+void SynchroSynth::carrierChanged()
+{
+	for (int i = 0; i < SYNCHRO_GRAPH_SAMPLES; ++i)
+	{
+		float phase = (float)i / (float)SYNCHRO_GRAPH_SAMPLES * F_2PI;
```

Comment:
More C-style casts to avoid here. However, instead of using `static_cast`, you could just reorder the operands here, e.g.
`F_2PI * i / SYNCHRO_GRAPH_SAMPLES` or `i * F_2PI / SYNCHRO_GRAPH_SAMPLES`.